core/src/avm2/globals.rs

```diff
@@ -658,10 +614,24 @@ pub fn load_player_globals<'gc>(
 
     // Inside this call, the macro `avm2_system_classes_playerglobal`
     // triggers classloading. Therefore, we run `load_playerglobal`
-    // relative late, so that it can access classes defined before
+    // relatively late, so that it can access classes defined before
     // this call.
     load_playerglobal(activation, domain)?;
 
+    // Except for `trace`, top-level builtin functions are defined
+    // on the `global` object.
+    define_fn_on_global(activation, "", "decodeURI", script);
```

Comment:
Not for this PR, just note for the future:

I think the ideal solution would be for the avm2.toplevel_global_object() and the global used when loading the script /globals/Toplevel.as to be the same object immediately, instead of copying the stuff after the fact.